analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/OutputFileHandler.kt

```diff
@@ -0,0 +1,47 @@
+package de.maibornwolff.codecharta.serialization
+
+import java.io.BufferedWriter
+import java.io.File
+import java.io.FileWriter
+import java.io.OutputStreamWriter
+import java.io.PrintStream
+import java.io.Writer
+
+object OutputFileHandler {
+
+    fun checkAndFixFileExtension(outputName: String): String {
+        if (outputName.isEmpty()) return "default.cc.json"
+        val delimiter = extractDelimiter(outputName)
+        if (delimiter.isEmpty()) {
+            return outputName.substringBefore(".") + ".cc.json"
+                }
+        return extractPath(outputName, delimiter) +
+               extractFileName(outputName, delimiter)
+    }
+
+     fun writer(outputName: String?, test: Boolean, output: PrintStream): Writer {
+         return if (test) {
+             OutputStreamWriter(output)
+         } else {
+             BufferedWriter(FileWriter(File(checkAndFixFileExtension(outputName ?: ""))))
+         }
+     }
+
+    private fun extractPath(outputName: String, delimiter: String): String {
+        return outputName.split(delimiter).dropLast(1).joinToString(delimiter) + delimiter
```

Comment:
I would prefere existing Utility functions (examples: https://mkyong.com/java/how-to-get-the-filepath-of-a-file-in-java/).
Thus, different operating system depending pathes are handled automatically.